core/source/Test/Context.mint

```diff
@@ -49,21 +59,33 @@ module Test.Context {
     test {
       with Test.Context {
         of(5)
-        |> Test.assertEqual(5)
+        |> assertEqual(5)
       }
     }
   */
-  fun assertEqual (a : a, context : Test.Context(a)) : Test.Context(a) {
+  fun assertEqual (value : a, context : Test.Context(a)) : Test.Context(a) {
     `
-    #{context}.step((subject) => {
-      if (!_compare(#{a}, subject)) {
-        throw \`Assertion failed: ${#{a}} === ${subject}\`
+    #{context}.step((subject)=> {
+      let result = _compare(#{value}, subject)
+
+      if (result) {
+        return subject
+      } else {
+        throw \`Assertion failed ${#{value}} === ${subject}\`
       }
-      return subject
```

Comment:
Please, put all of the changes to the `Context` module into a different PR instead.

Reply:
I've opened up #460 to address this.